freud/data.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2010-2019 The Regents of the University of Michigan
+# This file is from the freud project, released under the BSD 3-Clause License.
+
+R"""
+The :class:`freud.data` module provides certain sample data sets and utility
+functions that are useful for testing and examples.
+"""
+
+import numpy as np
+import freud
+
+
+class UnitCell(object):
+    """Class to represent the unit cell of a crystal structure.
+
+    This class represents the unit cell of a crystal structure, which is
+    defined by a lattice and a basis. It provides the basic attributes of the
+    unit cell as well as enabling the generation of systems of points
+    (optionally with some noise) from the unit cell.
+
+    Args:
+        box:
+            A box-like object (see :meth:`~freud.box.Box.from_box`) containing
+            the lattice vectors of the unit cell.
+        basis_positions ((:math:`N_{points}`) :class:`numpy.ndarray`):
+            The basis of the unit cell in fractional coordinates
+            (Default value = ``[[0, 0, 0]]``.
+    """
+    def __init__(self, box, basis_positions=[[0, 0, 0]]):
+        self._box = freud.box.Box.from_box(box)
+        self._basis_positions = np.dot(basis_positions, self.box.to_matrix().T)
+
+    def to_system(self, num_replicas=1, scale=1, sigma_noise=0):
+        """Generate a system from the unit cell.
+
+        Args:
+            num_replicas (:class:`tuple` or `int`):
+                If provided as a single number, the number of replicas in all
+                dimensions. If a tuple, the number of times replicated in each
+                dimension. Must be of the form (nx, ny, 1) for 2D boxes
+                (Default value = 1).
+            scale (float): Amount to scale the unit cell by (in distance units)
```

Comment:
The multiplicative factor is unitless, not in distance units...
```suggestion
            scale (float): Factor by which to scale the unit cell
```